core/operation/src/main/java/uk/gov/gchq/gaffer/operation/Operation.java

```diff
@@ -182,7 +188,17 @@ default void close() throws IOException {
      */
     default ValidationResult validate() {
         final ValidationResult result = new ValidationResult();
-        for (final Field field : getClass().getDeclaredFields()) {
+
+        HashSet<Field> fields = Sets.<Field>newHashSet();
+        Class<?> currentClass = this.getClass();
+        String packageName;
+        do {
+            fields.addAll(Arrays.asList(currentClass.getDeclaredFields()));
+            currentClass = currentClass.getSuperclass();
+            packageName = currentClass.getPackage().getName();
+        } while (packageName.startsWith(UK_GOV_GCHQ_GAFFER));
```

Comment:
I don't think we should limit it to the Gaffer package. Can we just look at all super classes? Just keep iterating up until currentClass is null.

Reply:
I thought that was excessive.